### ISSUE TITLE
Make Icon Only Share Links Accessible

### ISSUE THREAD

**Issue by davidakennedy** [Automattic/jetpack#1069]
Currently, if a site owner uses the "Icon Only" sharing links, they are not accessible to screen readers.

Current markup:

``` html
<div class="sd-content">
  <ul>
    <li class="share-twitter">
      <a rel="nofollow" class="share-twitter sd-button share-icon no-text" href="http://mysite.com/2014/01/01/hello-world/?share=twitter&amp;nb=1" title="Click to share on Twitter" id="sharing-twitter-1"><span></span>
      </a></li>
    <li class="share-end"></li></ul></div>
```

**Why?**
- These links use only icon fonts inside the `<a>`, but icon fonts are not announced by screen readers.
- The `title` attribute may be intended for screen readers, but it's not announced by default by screen readers either, and it's really only useful for users with a mouse. See this post for [more info on the `title` attribute](http://www.paciellogroup.com/blog/2013/01/using-the-html-title-attribute-updated/).
- When a screen reader announces these links, it says: "Link", which isn't descriptive and doesn't tell the user what the link does or where it goes.

**Suggestions**

Possible markup:

``` html
<div class="sd-content">
  <ul>
    <li class="share-twitter">
      <a rel="nofollow" class="share-twitter sd-button share-icon no-text" href="http://mysite.com/2014/01/01/hello-world/?share=twitter&amp;nb=1" title="Click to share on Twitter" id="sharing-twitter-1"><span class="screen-reader-text">Share on Twitter</span>
      </a></li>
    <li class="share-end"></li></ul></div>
```
- Links should always have link text.
- The `screen-reader-text` would be hidden off-screen, maintaining the design of these.
- Also ideally, if the links open in a new window, it would warn them of that. Like: "Share on Twitter (opens in new window)"

Let me know if there are any questions. Also, happy to test a solution. :smile: Thanks!
